src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -81,11 +94,45 @@ CHIP_ERROR OTAImageProcessorImpl::ProcessBlock(ByteSpan & block)
     VerifyOrReturnError(mDownloader != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
     CHIP_ERROR error = ProcessHeader(block);
-
     if (error == CHIP_NO_ERROR)
     {
-        // DFU target library buffers data internally, so do not clone the block data.
-        error = System::MapErrorZephyr(dfu_target_write(block.data(), block.size()));
+        mCurrentImage.mCurrentOffset += block.size();
+        if (mCurrentImage.mCurrentOffset >= mCurrentImage.mFileInfo->mFileSize)
+        {
+            // calculate how many data should be moved to the next image
+            uint64_t remainingDataSize = mCurrentImage.mCurrentOffset - static_cast<uint64_t>(mCurrentImage.mFileInfo->mFileSize);
+            // write last data of previous image
+            error = System::MapErrorZephyr(dfu_target_write(block.data(), block.size() - remainingDataSize));
+            // switch to net image
+            mCurrentImage.mIndex++;
+            mCurrentImage.mFileInfo = &mContentHeader.mFiles[mCurrentImage.mIndex];
+
+            if (OTAImageContentHeader::FileId::kNetMcuboot == mCurrentImage.mFileInfo->mFileId &&
+                mCurrentImage.mFileInfo->mFileSize > 0 && CHIP_NO_ERROR == error)
+            {
+                // finish previous image and reset target
+                dfu_target_done(true);
+                dfu_target_reset();
+                // initialize next dfu target to store net-core image.
+                dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, mCurrentImage.mIndex, /* size */ 0, nullptr);
```

Comment:
We can do it as a followup, but imho we should check the error code here. Perhaps, extracting this part of the code to a separate function would help to organize the code that way (you could do early returns on error).
Also, using `mCurrentImage.mIndex` is misleading as those values are only accidentally the same. In theory you could have network image on a different position in the OTA image. You can simply use `1` or some constant.

Reply:
You are right, this part can be moved to a separate function. mCurrentImage.mIndex contains information about current image which is not equal to mcuboot's image enumeration.  In mcuboot's image enum the mcuboot image is stored at 0, the app image is stored at 1 and the net core is stored at 2 so in this function it was needed to add another index. But I agree a constant will be better for our purpose because we don't support mcuboot image uptade yet.